nucypher/network/middleware.py

```diff
@@ -183,6 +182,41 @@ def method_wrapper(path,
 
         return method_wrapper
 
+    def _execute_method(self,
+                        node_or_sprout,
+                        host,
+                        port,
+                        method,
+                        endpoint,
+                        *args, **kwargs):
+        # Use existing cached SSL certificate or fetch fresh copy and retry
+        cached_certificate_filepath = Path(
+            self.storage.generate_certificate_filepath(host=host, port=port))
+        if cached_certificate_filepath.exists():
```

Comment:
LBYL -- Can this be done without checking for the cached cert first?  It will be nice to unnest as many blocks as possible here.

Reply:
The concern I have here is that passing a non-existent filepath will cause an `IOError` to be raised (https://github.com/psf/requests/blob/main/requests/adapters.py#L228). However, `IOError` is very general, and it won't be clear if the `IOError` is specifically because of the non-existent cert, or because of some other reason related to the actual execution of the method (if the cert was actually valid). I'm also hesitant to filter on text in the exception message...